controller/lib/core/sensors.cpp

```diff
@@ -71,8 +71,9 @@ AnalogPin pin_for(Sensor s) {
   __builtin_unreachable();
 }
 
-static float diameter_to_area(float diameter) {
-  return static_cast<float>(M_PI) / 4.0f * diameter * diameter;
+static float diameter_to_area(Length diameter) {
```

Comment:
Since we don't have a unit for square meters, should the function name reflect that? E.g. diameter_to_area_m2?

Reply:
Done